core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlDDLBuilder.scala

```diff
@@ -0,0 +1,42 @@
+package com.databricks.labs.remorph.parsers.tsql
+
+import com.databricks.labs.remorph.parsers.{OptionAuto, OptionExpression, OptionOff, OptionOn, OptionString, ParserCommon, intermediate => ir}
+
+import scala.collection.JavaConverters.asScalaBufferConverter
+
+class TSqlDDLBuilder extends TSqlParserBaseVisitor[ir.Catalog] with ParserCommon[ir.Catalog] {
+
+  private val optionBuilder = new OptionBuilder(new TSqlExpressionBuilder)
```

Comment:
inject via constructor instead of always initialising. this will be useful once we parametrise all those AST builders to support different sub versions of a dialect

Reply:
No problem.